### ISSUE TITLE
tpm2-tss and gcc

### ISSUE THREAD

**Issue by twoerner** [flihp/meta-measured#55]
tpm2-tss_1.2.0 (i.e. the default) doesn't build with gcc7 (i.e. the default gcc on master).

| In file included from ../tpm2-tss/include/sapi/tpm20.h:43:0,
|                  from ../tpm2-tss/include/tcti/tcti_socket.h:35,
|                  from ../tpm2-tss/tcti/sockets.cpp:4:
| ../tpm2-tss/include/sapi/implementation.h:159:50: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]

tpm2-tss_1.2.0 builds fine with gcc6 (from rocko, for example). Could a rocko branch be created that keeps the preferred version at 1.2.0 for rocko, then set the preferred version to "git%" for master?

**Comment by twoerner** [comment]
the same seems to be also true for:
- tpm2-abrmd
- tpm2-tools

**Comment by flihp** [comment]
Hey @twoerner. A rocko branch is a good idea. This is what we've done in the past, I'm just slacking off on my OE stuff a bit. I'll pick this up soon hopefully.

**Comment by twoerner** [comment]
Should master be switched to use the git versions by default? If nothing else, the SRCREVs on master should be bumped since gcc7 (the default on master) won't build the pinned SRCREVs without error.

**Comment by flihp** [comment]
Yeah that sounds about right. My autobuilder died last month and I haven't set it back up so my work on this is a bit stale.

**Comment by twoerner** [comment]
PS: the git version of tpm2-tss (5c3a67da97b1927f879c047f66c393090785f0ea) seems to have a dependency on uriparser.

A recipe for uriparser 0.8.4 can be found at https://layers.openembedded.org/layerindex/recipe/32523/

**Comment by flihp** [comment]
This dependency was added to the socket TCTI last week. Just haven't had the cycles to update the OE recipes. Probably worthy of a bug / issue ... or a PR :wink:

**Comment by twoerner** [comment]
I'm happy to provide a pull request, but the question I need answered is: how do you want to provide uriparser?
- copy+paste it from to meta-measured
- depend on meta-webos pr meta-luneui (least likely)
- submit it upstream to oe-core then wait for it to land before updating tpm2-tss's DEPENDs

**Comment by flihp** [comment]
Good question. First option is probably worst, last is best, the second will likely be a stop gap till oe-core takes the recipe. I've never sent a recipe upstream to oe-core, and definitely not a recipe from someone else's layer. Is there a "right way" to do this?

**Comment by twoerner** [comment]
I've emailed the owner of meta-luneui to ask if he would be interested in pushing his recipe upstream. I CC'ed you, hopefully he takes care of it then we can simply use it. In meta-luneui the uriparser is in a directory named "recipes-upstreamable", so there's hope...

**Comment by flihp** [comment]
Yeah that's a great sign. Thanks for pushing ahead on this.

**Comment by twoerner** [comment]
The owner of meta-luneui never got back to me, so I submitted uriparser to meta-openembedded. It looks like it's going to be added, we just need to wait for a "State of the World" cycle or two to pass before it gets in (they usually last a week or so each).

**Comment by flihp** [comment]
Forgot to add some relevant data to this thread: There's a rocko branch up now and I'm rolling versions forward for the tpm2-* recipes there. My current plan is to get that updated, merged back into master and then point master building from git. Gonna be a slow process I'm afraid with the current push to our 2.0 release of the core TSS2 libraries though.

**Comment by twoerner** [comment]
Thanks for creating the rocko branch :-)

This latest meta-openembedded cycle was a *lot* longer than usual due to some hardware issues. However, I just checked and the urparser recipe is now part of meta-openembedded. So I hope to create some pull requests this coming week.
